src/charm.py

```diff
@@ -242,6 +232,79 @@ def restart_mongos_service(self) -> None:
         self.stop_mongos_service()
         self.start_mongos_service()
 
+    def share_uri(self) -> None:
+        """Future PR - generate URI and give it to related app"""
+        # TODO future PR - generate different URI for data-integrator as that charm will not
+        # communicate to mongos via the Unix Domain Socket.
+
+    def set_user_role(self, roles: List[str]):
+        """Updates the roles for the mongos user."""
+        roles = roles.join(",")
```

Comment:
nits: in general it is better not "shadow" the function param, especially when the variable type will change , but create a new variable. something like `roles_strs` or rename `roles` to `roles_list` :)

Reply:
ah this is a VERY good tip, thank you Dmitry!!